crates/ere-risc0/src/prove/bento.rs

```diff
@@ -0,0 +1,230 @@
+use crate::{Risc0Program, SDK_VERSION};
+use bonsai_sdk::blocking::Client;
+use risc0_zkvm::{Receipt, VERSION, serde::to_vec};
+use std::{
+    env,
+    ffi::OsStr,
+    io::{self, Write},
+    process::{Command, Output, Stdio},
+    time::Duration,
+};
+use tempfile::tempdir;
+use zkvm_interface::{Input, InputItem, zkVMError};
+
+const URL: &str = "http://localhost:8081";
+const KEY: &str = "";
+
+// Copied and modified from https://github.com/risc0/risc0/blob/main/bento/crates/bento-client/src/bento_cli.rs.
+pub fn prove(program: &Risc0Program, inputs: &Input) -> Result<(Receipt, Duration), zkVMError> {
+    let client = Client::from_parts(URL.to_string(), KEY.to_string(), VERSION)
+        .map_err(|err| zkVMError::Other(err.into()))?;
+
+    // Serialize `inputs` in the same way `ExecutorEnv` does.
+    let mut input_bytes = Vec::new();
+    for input in inputs.iter() {
+        match input {
+            InputItem::Object(serialize) => {
+                input_bytes.extend(bytemuck::cast_slice(&to_vec(serialize).unwrap()));
+            }
+            InputItem::Bytes(items) => {
+                input_bytes.extend((items.len() as u32).to_le_bytes());
+                input_bytes.extend(items);
+            }
+        }
+    }
+
+    client
+        .upload_img(&program.image_id.to_string(), program.elf.clone())
+        .map_err(|err| zkVMError::Other(err.into()))?;
+    let input_id = client
+        .upload_input(input_bytes)
+        .map_err(|err| zkVMError::Other(err.into()))?;
+
+    let now = std::time::Instant::now();
+
+    let session = client
+        .create_session(program.image_id.to_string(), input_id, vec![], false)
+        .map_err(|err| zkVMError::Other(err.into()))?;
+
+    loop {
+        let res = session
+            .status(&client)
+            .map_err(|err| zkVMError::Other(err.into()))?;
+
+        match res.status.as_ref() {
+            "RUNNING" => {
+                std::thread::sleep(Duration::from_millis(200));
```

Comment:
By interval checking if a proof is ready or not, currently setting 200ms to not making the proving time too inaccurate, but I think in the future it'd be better to observe the container log or use other better way to do this.